test/conformance/ingress/headers_test.go

```diff
@@ -0,0 +1,304 @@
+		}
+
+		got, want := k.Request.Headers.Get(headerName), name
+		if got != want {
+			t.Errorf("Headers[%q] = %s, wanted %s", headerName, got, want)
+		}
+	})
+}
+
+// TestPostSplitSetHeaders verifies that an Ingress that specified AppendHeaders post-split has the appropriate header(s) set.
+func TestPostSplitSetHeaders(t *testing.T) {
+	t.Parallel()
+	clients := test.Setup(t)
+
+	const headerName = "Foo-Bar-Baz"
+
+	backends := make([]v1alpha1.IngressBackendSplit, 0, 10)
+	names := sets.NewString()
+	for i := 0; i < 10; i++ {
+		name, port, cancel := CreateService(t, clients, networking.ServicePortNameHTTP1)
+		defer cancel()
+		test.CleanupOnInterrupt(cancel)
+		backends = append(backends, v1alpha1.IngressBackendSplit{
+			IngressBackend: v1alpha1.IngressBackend{
+				ServiceName:      name,
+				ServiceNamespace: test.ServingNamespace,
+				ServicePort:      intstr.FromInt(port),
+			},
+			// Append different headers to each split, which lets us identify
+			// which backend we hit.
+			AppendHeaders: map[string]string{
+				headerName: name,
+			},
+			Percent: 10,
+		})
+		names.Insert(name)
+	}
+
+	// Create a simple Ingress over the 10 Services.
+	name := test.ObjectNameForTest(t)
+	ing := &v1alpha1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: test.ServingNamespace,
+			Annotations: map[string]string{
+				networking.IngressClassAnnotationKey: test.ServingFlags.IngressClass,
+			},
+		},
+		Spec: v1alpha1.IngressSpec{
+			Rules: []v1alpha1.IngressRule{{
+				Hosts:      []string{name + ".example.com"},
+				Visibility: v1alpha1.IngressVisibilityExternalIP,
+				HTTP: &v1alpha1.HTTPIngressRuleValue{
+					Paths: []v1alpha1.HTTPIngressPath{{
+						Splits: backends,
+					}},
+				},
+			}},
+		},
+	}
+	ing, err := clients.NetworkingClient.Ingresses.Create(ing)
+	if err != nil {
+		t.Fatalf("Error creating Ingress: %v", err)
+	}
+	defer clients.NetworkingClient.Ingresses.Delete(ing.Name, &metav1.DeleteOptions{})
+	test.CleanupOnInterrupt(func() { clients.NetworkingClient.Ingresses.Delete(ing.Name, &metav1.DeleteOptions{}) })
+
+	if err := v1a1test.WaitForIngressState(clients.NetworkingClient, ing.Name, v1a1test.IsIngressReady, t.Name()); err != nil {
+		t.Fatalf("Error waiting for ingress state: %v", err)
+	}
+	ing, err = clients.NetworkingClient.Ingresses.Get(name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Error getting Ingress: %v", err)
+	}
+
+	// Create a client with a dialer based on the Ingress' public load balancer.
+	client := &http.Client{
+		Transport: &http.Transport{
+			DialContext: CreateDialContext(t, ing, clients),
+		},
+	}
+
+	t.Run("Check without passing header", func(t *testing.T) {
+		// Make enough requests that the likelihood of us seeing each variation is high,
+		// but don't check the distribution of requests, as that isn't the point of this
+		// particular test.
+		seen := sets.NewString()
+		for i := 0; i < 100; i++ {
```

Comment:
Do those in parallel? Or at least in several goroutines?

Reply:
You are gonna hate my next test 😈.  I tried this there (where I'm issuing 1000), and I can't successfully run it on my macbook locally, so I reverted it.

I also thought better of hammering things too much because this isn't a ksvc, but a single Pod 😬

Reply:
Hence the second "several go routines" :) We can run 10x10 probably :)

Reply:
Yeah, I'd want to adopt our `pkg/pool` for this, however, the `errgroup` pattern I'd want to adopt is actually the context-based variety, so I can `select` on the error channel or a channel of seen headers, and we don't have that currently (wanna PR it? :D).